File: WebClient SDK/src/default/workflow/krProcess/formatters/editStageTypeFormatter.ts
import { extension } from '@tessa/application';
import {
  IKrStageTypeFormatterContext,
  KrStageTypeFormatter,
  KrStageTypeFormattingHelper,
  StageTypeDescriptor,
  StageTypeDescriptors
} from '@tessa/platform';

@extension({ name: 'EditStageTypeFormatter' })
export class EditStageTypeFormatter extends KrStageTypeFormatter {
  //#region constants and static fields

  private static readonly _changeState = KrStageTypeFormattingHelper.formatPlainColumnName(
    'KrEditSettingsVirtual',
    'ChangeState'
  );

  //#endregion

  //#region base overrides

  descriptors(): ReadonlyArray<StageTypeDescriptor> {
    return [StageTypeDescriptors.editDescriptor];
  }

  format(context: IKrStageTypeFormatterContext): void {
    super.format(context);
    context.displaySettings = context.stageRow.tryGet(EditStageTypeFormatter._changeState)
      ? '$UI_KrEdit_ChangeState'
      : '';
  }

  //#endregion
}

File: Extensions/Tessa.Extensions.Default.Shared/Workflow/KrProcess/Formatters/EditStageTypeFormatter.cs
#nullable enable

using System.Threading.Tasks;
using Tessa.Platform.Storage;

namespace Tessa.Extensions.Default.Shared.Workflow.KrProcess.Formatters
{
    /// <summary>
    /// Форматтер типа этапа <see cref="StageTypeDescriptors.EditDescriptor"/>.
    /// </summary>
    public sealed class EditStageTypeFormatter :
        StageTypeFormatterBase
    {
        #region Base Overrides

        /// <inheritdoc />
        public override async ValueTask FormatClientAsync(
            IStageTypeFormatterContext context)
        {
            await base.FormatClientAsync(context);

            context.DisplaySettings = context.StageRow.Fields.Get<bool>(KrConstants.KrEditSettingsVirtual.ChangeState)
                ? "$UI_KrEdit_ChangeState"
                : string.Empty;
        }

        /// <inheritdoc />
        public override async ValueTask FormatServerAsync(
            IStageTypeFormatterContext context)
        {
            await base.FormatServerAsync(context);

            context.DisplaySettings = context.Settings!.TryGet<bool>(KrConstants.KrEditSettingsVirtual.ChangeState)
                ? "$UI_KrEdit_ChangeState"
                : string.Empty;
        }

        #endregion
    }
}
